src/restic/node.go

```diff
@@ -18,36 +18,29 @@ import (
 	"runtime"
 )
 
-// ExtendedAttribute is a tuple storing the xattr name and value.
-type ExtendedAttribute struct {
-	Name  string `json:"name"`
-	Value []byte `json:"value"`
-}
-
 // Node is a file, directory or other item in a backup.
 type Node struct {
-	Name               string              `json:"name"`
-	Type               string              `json:"type"`
-	Mode               os.FileMode         `json:"mode,omitempty"`
-	ModTime            time.Time           `json:"mtime,omitempty"`
-	AccessTime         time.Time           `json:"atime,omitempty"`
-	ChangeTime         time.Time           `json:"ctime,omitempty"`
-	UID                uint32              `json:"uid"`
-	GID                uint32              `json:"gid"`
-	User               string              `json:"user,omitempty"`
-	Group              string              `json:"group,omitempty"`
-	Inode              uint64              `json:"inode,omitempty"`
-	Size               uint64              `json:"size,omitempty"`
-	Links              uint64              `json:"links,omitempty"`
-	LinkTarget         string              `json:"linktarget,omitempty"`
-	ExtendedAttributes []ExtendedAttribute `json:"extended_attributes,omitempty"`
-	Device             uint64              `json:"device,omitempty"`
-	Content            IDs                 `json:"content"`
-	Subtree            *ID                 `json:"subtree,omitempty"`
+	Name               string            `json:"name"`
+	Type               string            `json:"type"`
+	Mode               os.FileMode       `json:"mode,omitempty"`
+	ModTime            time.Time         `json:"mtime,omitempty"`
+	AccessTime         time.Time         `json:"atime,omitempty"`
+	ChangeTime         time.Time         `json:"ctime,omitempty"`
+	UID                uint32            `json:"uid"`
+	GID                uint32            `json:"gid"`
+	User               string            `json:"user,omitempty"`
+	Group              string            `json:"group,omitempty"`
+	Inode              uint64            `json:"inode,omitempty"`
+	Size               uint64            `json:"size,omitempty"`
+	Links              uint64            `json:"links,omitempty"`
+	LinkTarget         string            `json:"linktarget,omitempty"`
+	ExtendedAttributes map[string][]byte `json:"extended_attributes,omitempty"`
```

Comment:
But this is a matter of representation in memory, not on disk. For one, in my experience there are only a few (like max 10 or so) different extended attributes in active use, so even n^2 is still small. Second, if that's a concern, then we can also have a `map[string][]byte` for the extended attributes in the `file` struct, here: https://github.com/restic/restic/blob/master/src/restic/fuse/file.go#L31

Did you measure the impact of this change? Is it worth it?